docker/images/n8n/Dockerfile

```diff
@@ -10,15 +10,21 @@ RUN --mount=type=cache,id=pnpm-store,target=/root/.local/share/pnpm/store --moun
 RUN pnpm build
 
 # Delete all dev dependencies
-RUN jq 'del(.pnpm.patchedDependencies)' package.json > package.json.tmp; mv package.json.tmp package.json
 RUN node .github/scripts/trim-fe-packageJson.js
+# We don't want to remove all patches because we want them still to be applied
+# in `pnpm deploy`. However, we need to remove FE patches because we trim the FE
+# package.json files and `pnpm deploy` will fail otherwise. element-plus is the
+# only FE patch that we need to remove.
+RUN jq 'del(.pnpm.patchedDependencies."element-plus@2.4.3")' package.json > package.json.tmp; mv package.json.tmp package.json
 
 # Delete any source code or typings
 RUN find . -type f -name "*.ts" -o -name "*.vue" -o -name "tsconfig.json" -o -name "*.tsbuildinfo" | xargs rm -rf
 
 # Deploy the `n8n` package into /compiled
 RUN mkdir /compiled
-RUN NODE_ENV=production DOCKER_BUILD=true pnpm --filter=n8n --prod --no-optional --legacy deploy /compiled
+# We don't want to use --no-optional because pdfjs-dist has an optional dependency on
+# @napi-rs/canvas, which is required for the pdfjs-dist package to work.
+RUN NODE_ENV=production DOCKER_BUILD=true pnpm --filter=n8n --prod --legacy deploy /compiled
```

Comment:
have you tested this? when I tried `@napi/canvas` did get installed, but failed to load because of missing binaries.

Reply:
I did, and it did work after this. It seemed that the package was indeed missing. But the binary was also missing unless I specified `linux/amd64` platform specifically. I can try and see if it's needed

Reply:
Okay tested it with the `--no-optional` and it does not work. So we need to remove it